tests/unit/test_advanced.py

```diff
@@ -943,3 +939,72 @@ def test_do_attempt_correct_takes_decoy_into_account(self):
         res = self._do_attempt()
 
         self.assertFalse(res['correct'])
+
+    def test_show_answer_status_always(self):
+        """
+         Test "Show Answer" button is shown in assessment mode when course "showanswer" setting is always
+        """
+        self.block.showanswer = SHOWANSWER.ALWAYS
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+
+        self.assertEqual(res.status_code, 200)
+
+    def test_show_answer_status_never(self):
+        """
+         Test "Show Answer" button is not shown in assessment mode when course "showanswer" setting is never
+        """
+        self.block.showanswer = SHOWANSWER.NEVER
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+
+        self.assertEqual(res.status_code, 400)
+
+    def test_show_answer_status_attempted(self):
+        """
+         Test "Show Answer" button is shown in assessment mode when course "showanswer" setting is attempted
+        """
+        self.block.showanswer = SHOWANSWER.ATTEMPTED
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+
+        self.assertEqual(res.status_code, 400)
+
+        self._do_attempt()
+
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+
+        self.assertEqual(res.status_code, 200)
+
+    def test_show_answer_status_is_correct(self):
+        """
+         Test "Show Answer" button is shown in assessment mode when course "showanswer" setting is ANSWERED
+        """
+        self.block.showanswer = SHOWANSWER.ANSWERED
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+
+        self.assertEqual(res.status_code, 400)
+
+        self._submit_complete_solution()
+
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+
+        self.assertEqual(res.status_code, 200)
+
+    def test_show_answer_status_is_finished(self):
+        """
+         Test "Show Answer" button is shown in assessment mode when course "showanswer" setting is FINISHED
+        """
+        self.block.showanswer = SHOWANSWER.FINISHED
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+
+        self.assertEqual(res.status_code, 400)
+
+        self._submit_incorrect_solution()
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+        self.assertEqual(res.status_code, 400)
+
+        self._submit_complete_solution()
+        res = self.call_handler(self.SHOW_ANSWER_HANDLER, data={}, expect_json=False)
+        self.assertEqual(res.status_code, 200)
+
+        self.assertTrue(self.block.attempts_remain)
+
+        self.assertFalse(self.block.closed)
```

Comment:
@abodacs The quality tests are failing here because the line count is exceeding 1000. Can you kindly see if you can remove some of the empty lines at appropriate placed to get it to stay under 1000?